src/core/MemoryModel/Storage/Serialization/TaskDeserializer.cpp

```diff
@@ -0,0 +1,23 @@
+//
+// Created by rodion on 9/26/20.
+//
+
+#include "TaskDeserializer.h"
+
+std::shared_ptr<TaskEntity> TaskDeserializer::DeserializeTask(const TaskProto &task) {
+  auto task_ = Task::create(task.name(), task.label(),
+                            TaskDeserializer::DeserializePriority(task.priority()),
+                            boost::gregorian::date{task.deadline()});
+  if(!task_.has_value()) return nullptr;
```

Comment:
why don't you use "Optional" as return value? 
someone, who call this method, will check for nullptr.
Reads better "has_value".